### DISCUSSION TITLE
How does it work when a server module imports the routes?

### DISCUSSION THREAD

**Discussion by yahiakala** [anvil-works/routing#46] [Q&A]
This line in the docs :

## Server Routes Module
Inside your Server code, create a server module that imports the routes module.

```python
# ServerRoutes.py
from . import routes
```

How does this actually work? Does it import the routes module from the client side?

**Comment by s-cork** [comment]
yes
In anvil you can import client modules from the server (otherwise portable classes wouldn't work)

When you do this on the server
the `Route` base class will create `anvil.server.route` callables for all your routes that define paths


```python
# Pseudo code
class Route:
    def __init_subclass__(cls, *args, **kws):
        if anvil.is_server_side() and cls.path is not None:
            @anvil.server.route(cls.path)
            def route_handler(*args, **kws):
                # serve the route from the server
                ...

```
            



This code can be found here:
https://github.com/anvil-works/routing/blob/f7645aab565e50db8a40ad929048c1ab49c2e2ee/client_code/router/_route.py#L24-L101

which is called at the end of the `__init_subclass__` method:
https://github.com/anvil-works/routing/blob/f7645aab565e50db8a40ad929048c1ab49c2e2ee/client_code/router/_route.py#L173-L208
